package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,599 @@
+            Bonds are found using a distance search, if unit cell information
+            is given, periodic boundary conditions will be considered in the
+            distance search. [``None``]
+
+        Returns
+        -------
+        list
+            List of tuples suitable for use in Universe topology building.
+
+        Warnings
+        --------
+        No check is done after the bonds are guessed to see if Lewis
+        structure is correct. This is wrong and will burn somebody.
+
+        Raises
+        ------
+        :exc:`ValueError` if inputs are malformed or `vdwradii` data
+        is missing.
+
+
+        .. _`same algorithm that VMD uses`:
+           http://www.ks.uiuc.edu/Research/vmd/vmd-1.9.1/ug/node26.html
+
+        """
+        # why not just use atom.positions?
+        if len(atoms) != len(coords):
+            raise ValueError("'atoms' and 'coord' must be the same length")
+
+        fudge_factor = kwargs.get('fudge_factor', 0.55)
+
+        # so I don't permanently change it
+        vdwradii = tables.vdwradii.copy()
+        user_vdwradii = kwargs.get('vdwradii', None)
+        # this should make algo use their values over defaults
+        if user_vdwradii:
+            vdwradii.update(user_vdwradii)
+
+        # Try using types, then elements
+        atomtypes = atoms.types
+
+        # check that all types have a defined vdw
+        if not all(val in vdwradii for val in set(atomtypes)):
+            raise ValueError(("vdw radii for types: " +
+                              ", ".join([t for t in set(atomtypes) if
+                                         t not in vdwradii]) +
+                              ". These can be defined manually using the" +
+                              " keyword 'vdwradii'"))
+
+        lower_bound = kwargs.get('lower_bound', 0.1)
+
+        if box is not None:
+            box = np.asarray(box)
+
+        # to speed up checking, calculate what the largest possible bond
+        # atom that would warrant attention.
+        # then use this to quickly mask distance results later
+        max_vdw = max([vdwradii[t] for t in atomtypes])
+
+        bonds = []
+
+        pairs, dist = distances.self_capped_distance(coords,
+                                                     max_cutoff=2.0 * max_vdw,
+                                                     min_cutoff=lower_bound,
+                                                     box=box)
+        for idx, (i, j) in enumerate(pairs):
+            d = (vdwradii[atomtypes[i]] +
+                 vdwradii[atomtypes[j]]) * fudge_factor
+            if (dist[idx] < d):
+                bonds.append((atoms[i].index, atoms[j].index))
+        return tuple(bonds)
+
+    def add_guessed_bonds(self):
+        vdwradii = self._kwargs.get('vdwradii', None)
+        if not hasattr(self._universe.atoms, 'types'):
+            self._universe.guess_TopologyAttributes(
+                context=self.context, to_guess=['types'])
```

Comment:
Again, it looks like calling this function will have the potentially unexpected side-effect of guessing types on the base Universe, which could be surprising -- is there any way to do this without changing the Universe? Perhaps by passing `atomtypes` in as an argument to `guess_bonds`?